pkg/controllers/jobset_controller.go

```diff
@@ -293,6 +261,158 @@ func (r *JobSetReconciler) createHeadlessSvcIfNotExist(ctx context.Context, js *
 	return nil
 }
 
+func (r *JobSetReconciler) executeFailurePolicy(ctx context.Context, js *jobset.JobSet, ownedJobs *childJobs) error {
+	// If no failure policy is defined, the default failure policy is to mark the JobSet
+	// as failed if any of its jobs have failed.
+	if js.Spec.FailurePolicy == nil {
+		return r.updateStatus(ctx, js, metav1.Condition{
+			Type:    string(jobset.JobSetFailed),
+			Status:  metav1.ConditionStatus(corev1.ConditionTrue),
+			Reason:  "FailedJobs",
+			Message: "jobset failed due to one or more failed jobs",
+		})
```

Comment:
r.updateStatusWithCondition(ctx, js, v1.EventTypeWarning, metav1.Condition{
			Type:    string(jobset.JobSetFailed),
			Status:  metav1.ConditionStatus(corev1.ConditionTrue),
			Reason:  "FailedJobs",
			Message: "jobset failed due to one or more failed jobs",
		})